enum.rb

```diff
@@ -0,0 +1,159 @@
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    ar = self
+    ar.each do |el|
+      yield el
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    i = 0
+    ar = self
+    ar.my_each do |element|
+      yield i, element
+      i += 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    ar = self
+    new_arr = []
+    ar.my_each do |i|
+      next unless (yield i) == true
+
+      new_arr.append(i)
+    end
+    new_arr
+  end
+
+  def my_all?(proc = nil)
+    ar = self
+    return ar.all?(proc) if proc
+
+    if block_given?
+      ar.my_each do |i|
+        next unless (yield i) == false
+
+        return false
+      end
+    end
+
+    return true if ar.my_count.zero?
+
+    !(!ar[0])
+  end
+
+  def my_none?(proc = nil)
+    ar = self
+    return ar.none?(proc) if proc
+
+    if block_given?
+      ar.my_each do |i|
+        next unless (yield i) == true
+
+        return false
+      end
+    end
+    !ar[ar.length - 1]
+  end
+
+  def my_any?(proc = nil)
+    ar = self
+    return ar.any?(proc) if proc
+
+    if block_given?
+      ar.my_each do |i|
+        next unless (yield i) == true
+
+        return true
+      end
+      return false
+    end
+
+    true
+  end
+
+  def my_count(proc = nil)
+    ar = self
+    i = 0
+    if proc
+      ar.my_each do |x|
+        next unless x == proc
+
+        i += 1
+      end
+      return i
+    end
+    if block_given?
+      ar.my_each do |x|
+        next unless (yield x) == true
+
+        i += 1
+      end
+      return i
+    end
+
+    ar.my_each do
+      i += 1
+    end
+    i
+  end
+
+  def my_map
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns an Enumerator if no block is given
```ruby
array = Array.new(100){rand(0...9)}
array.my_map == Enumerator #true
```

Reply:
### This edge case is still failing.